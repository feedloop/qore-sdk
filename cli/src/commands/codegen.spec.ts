import { setupRecorder } from "nock-record";
import fs from "fs";
import path from "path";
import Codegen from "./codegen";

const record = setupRecorder();

describe("codegen", () => {
  it("should be able to generate codegen", async () => {
    process.env.REACT_APP_API_URL =
      "https://p-qore-dot-pti-feedloop.et.r.appspot.com";
    const { completeRecording } = await record("codegen");
    await Codegen.run([
      "--project",
      "I0D3NimZQ9GKEDP",
      "--org",
      "lIdfC42DJCN2XzQ",
      "--token",
      "77f2ff71-8864-404d-8596-127d78a4c1bd",
    ]);
    completeRecording();
    const filename = path.resolve(process.cwd() + "/qore-generated.ts");
    expect(
      fs.readFileSync(filename, {
        encoding: "utf8",
      })
    ).toMatchSnapshot();
    fs.unlinkSync(filename);
  });
});
