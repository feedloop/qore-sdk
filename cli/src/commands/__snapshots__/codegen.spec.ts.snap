// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`codegen should be able to generate codegen 1`] = `
"type MemberTableRow = {
  id: string;
  role: { id: string; displayField: string };
  email: string;
  toDo: ToDoTableRow[];
  totalTask: number;
  totalPoint: number;
  totalHardTask: number;
  createTask: action;
};

type ToDoTableRow = {
  id: string;
  task: string;
  description: string;
  done: boolean;
  points: number;
  deadline: date;
  difficulty: select;
  person: PersonTableRow[];
  role: lookup;
  attachment: file;
  timeAllocation: formula;
};

type MemberDefaultViewViewRow = {
  read: {
    id: string;
    email: string;
    role: { id: string; displayField: string };
    createTask: action;
  };
  write: {
    email: string;
  };
  params: {};
};

type ToDoDefaultViewViewRow = {
  read: {
    id: string;
    task: string;
    done: boolean;
    points: number;
    deadline: date;
    difficulty: select;
    person: PersonTableRow[];
    role: lookup;
    attachment: file;
    timeAllocation: formula;
  };
  write: {
    task: string;
    done: boolean;
    points: number;
    deadline: date;
    difficulty: select;
    person: string[];
    attachment: file;
  };
  params: {
    \\"$by.points\\"?: \\"desc\\";
  };
};

type TestViewRow = {
  read: {
    id: string;
    task: string;
    description: string;
  };
  write: {
    task: string;
    description: string;
  };
  params: {};
};

type ToDoViewRow = {
  read: {
    id: string;
    task: string;
    description: string;
    done: boolean;
    points: number;
    deadline: date;
  };
  write: {
    task: string;
    description: string;
    done: boolean;
    points: number;
    deadline: date;
  };
  params: {};
};

type DoneViewRow = {
  read: {
    id: string;
    task: string;
    description: string;
    done: boolean;
    points: number;
    deadline: date;
  };
  write: {
    task: string;
    description: string;
    done: boolean;
    points: number;
    deadline: date;
  };
  params: {};
};

type ReisTaskViewRow = {
  read: {
    id: string;
    task: string;
    description: string;
    done: boolean;
    points: number;
    deadline: date;
    difficulty: select;
    person: PersonTableRow[];
    role: lookup;
    attachment: file;
    timeAllocation: formula;
  };
  write: {
    task: string;
    description: string;
    done: boolean;
    points: number;
    deadline: date;
    difficulty: select;
    person: string[];
    attachment: file;
  };
  params: {
    timeAllocation?: string;
  };
};

type MemberMetricViewRow = {
  read: {
    id: string;
    email: string;
    role: { id: string; displayField: string };
    totalTask: number;
    totalPoint: number;
    totalHardTask: number;
  };
  write: {
    email: string;
  };
  params: {
    email: string;
  };
};

export type QoreProjectSchema = {
  memberDefaultView: MemberDefaultViewViewRow;
  toDoDefaultView: ToDoDefaultViewViewRow;
  test: TestViewRow;
  toDo: ToDoViewRow;
  done: DoneViewRow;
  reisTask: ReisTaskViewRow;
  memberMetric: MemberMetricViewRow;
};
"
`;
